linkedlist.c

```diff
@@ -0,0 +1,71 @@
+//erzeugt listenelement
+int* create(int* next, int data) {
+	int* s;
+	s = (int*)malloc( 2 * 4 );
+
+	// next == 0 entspricht dem letzten element, ansonsten pointer aufs naechste
+	*(s+0) = next;
+
+	// payload, daten kommen hier rein, zB 'a'
+	*(s+1) = data;
+
+	return s;
+}
+
+// entfernt element von liste
+// sucht element in liste rekursiv
+void remove(int* head, int* element) {
```

Comment:
it's just a little easier to read and (thus) more "elegant" - but: of course you can iterate through the list.